### PR TITLE
add nRF52840 based Makerdiary Pitaya Go

### PR DESCRIPTION
**Pull request by xiongyihui**
Add [a new board Pitaya Go from Makerdiary](https://github.com/makerdiary/pitaya-go)

### CODE REVIEW

File: ports/nrf/boards/pitaya_go/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,10 @@
+USB_VID = 0x239A
+USB_PID = 0x802A
```
Comment (ladyada) [inline ports/nrf/boards/pitaya_go/mpconfigboard.mk:2]: Do not recycle VID/PID, they must be unique for each device!

Comment (xiongyihui) [reply ports/nrf/boards/pitaya_go/mpconfigboard.mk:2]: OK. As Makerdiary doesn't have a VID, [maybe use a random generated PID with NordicSemi's VID](https://devzone.nordicsemi.com/f/nordic-q-a/50638/usb-pid-for-nrf52840)

Comment (ladyada) [reply ports/nrf/boards/pitaya_go/mpconfigboard.mk:2]: hi, as you make many products with USB, please purchase a VID!
https://store.makerdiary.com/

Comment (xiongyihui) [reply ports/nrf/boards/pitaya_go/mpconfigboard.mk:2]: can not afford the licensing fee yet

**Pull request merged**